backend/tracim_backend/lib/core/user.py

```diff
@@ -201,15 +201,44 @@ def get_known_users(
         :param exclude_user_ids: user id to exclude from result
         :param exclude_workspace_ids: workspace user to exclude from result
         :param include_workspace_ids: only include users from these workspaces
-        :nb_elem: number of users to return, default value should be low for
-        security and privacy reasons
+        :limit: maximum number of users to return. This value will be capped to
+                KNOWN_MEMBERS_ITEMS_LIMIT if requesting users from workspaces that
+                the requester is not part of.
         :filter_results: If true, do filter result according to user workspace if user is provided
         :return: List of found users
         """
 
-        if len(acp) < 2:
+        nb_elems = KNOWN_MEMBERS_ITEMS_LIMIT
+
+        # RJ - 2020-09-14-NOTE
+        # This is a bit convoluted. Keep in mind that we want to allow:
+        #  - listing a number of members greater than KNOWN_MEMBERS_ITEMS_LIMIT
+        #  - autocomplete less than 2 characters
+        # only if the requesting user is in each and every included workspace.
+        # Otherwise, we don't want to allow that.
+        # By default, we return a maximum of KNOWN_MEMBERS_ITEMS_LIMIT members (when limit is not set),
+
+        user_in_every_included_workspaces = False  # type: bool
+        include_user_ids = None  # type: typing.Optional[typing.List[int]]
+
+        if include_workspace_ids:
+            user_in_every_included_workspaces = True
+
+            include_user_ids = []
+            for workspace_id in include_workspace_ids:
+                user_ids = self.get_members_of_workspaces([workspace_id])
+                include_user_ids += user_ids
+                if user_in_every_included_workspaces:
+                    user_in_every_included_workspaces = self._user.user_id in user_ids
```

Comment:
I'm afraid this will be a bit slow if many workspace ids are given. Wouldn't a single query on `UserRoleInWorkspace` work?
```python
query(UserRoleInWorkspace).filter(UserRoleInWorkspace.user_id==self._user.user_id).filter(UserRoleInWorkspace.workspace_id.in_(include_workspace_ids)).count()
```
then compare the count with the count of workspaces?

Reply:
Good idea

Reply:
After discussion, we decided not to do this for now: we plan to refactor this method. Moreover, in the current use case, only one space is ever listed in include_workspaces, so this would add a query and make the method slower instead of making it faster.